great_expectations/data_asset/data_asset.py

```diff
@@ -17,7 +17,7 @@
 
 from great_expectations.version import __version__
 from great_expectations.data_asset.util import DotDict, recursively_convert_to_json_serializable, parse_result_format
-from great_expectations.dataset.autoinspect import columns_exist
+from great_expectations.profile.autoinspect import columns_exist
```

Comment:
Backwards compatibility issues aside, I think picking either profile or autoinspect is preferable to using both unless you're imagining a substantive semantic difference between them. I'd prefer to always express semantically meaningful differences in confidence in the "confidence" flag.

Reply:
:+1:

I consider `from great_expectations.profile.autoinspect import columns_exist` a deprecated use case.

At this point, we can fully replace with `from great_expectations.profile.columns_exist import ColumnsExistProfiler`

I'll make that change.

Reply:
Hm. I ended up banging my head on this one for quite a while. 

Naively switching to ColumnsExistProfiler creates an import loop:
```
$ pytest tests/test_data_asset.py 
ImportError while loading conftest '/Users/abe/Documents/superconductive/tools/great_expectations/tests/conftest.py'.
tests/conftest.py:10: in <module>
    import great_expectations as ge
great_expectations/__init__.py:1: in <module>
    from .util import *
great_expectations/util.py:11: in <module>
    import great_expectations.dataset as dataset
great_expectations/dataset/__init__.py:3: in <module>
    from .dataset import Dataset
great_expectations/dataset/dataset.py:18: in <module>
    from great_expectations.data_asset.data_asset import DataAsset
great_expectations/data_asset/__init__.py:1: in <module>
    from .data_asset import DataAsset
great_expectations/data_asset/data_asset.py:20: in <module>
    from great_expectations.profile.columns_exist import ColumnsExistProfiler
great_expectations/profile/columns_exist.py:1: in <module>
    from .base import DataSetProfiler
great_expectations/profile/base.py:4: in <module>
    from ..dataset import Dataset
E   ImportError: cannot import name 'Dataset'
```

Reply:
As discussed, let's just pull support for legacy autoinspect at this point. I updated everything to use ColumnsExistProfiler, and simply removed the default profiler.